src/statementsDiagnostics/store/statementDiagnosticsReports.reducer.ts

```diff
@@ -0,0 +1,166 @@
+import { Action, Reducer } from "redux";
+import { updateLocalStorage } from "../../store/localStorage";
+import { ActionWithPayload } from "../../store";
+
+export const STATEMENTS_DIAGNOSTICS_REFRESH =
+  "ADMIN_UI/STATEMENTS_DIAGNOSTICS_REFRESH";
+export const STATEMENTS_DIAGNOSTICS_REQUEST =
+  "ADMIN_UI/STATEMENTS_DIAGNOSTICS_REQUEST";
+export const STATEMENTS_DIAGNOSTICS_FAILED =
+  "ADMIN_UI/STATEMENTS_DIAGNOSTICS_FAILED";
+export const STATEMENTS_DIAGNOSTICS_RECEIVED =
+  "ADMIN_UI/STATEMENTS_DIAGNOSTICS_RECEIVED";
+export const STATEMENTS_DIAGNOSTICS_INVALIDATED =
+  "ADMIN_UI/STATEMENTS_DIAGNOSTICS_INVALIDATED";
+export const STATEMENT_DIAGNOSTICS_CREATE_REPORT =
+  "ADMIN_UI/STATEMENT_DIAGNOSTICS_CREATE_REPORT";
+export const STATEMENT_DIAGNOSTICS_CREATE_REPORT_COMPLETED =
+  "ADMIN_UI/STATEMENT_DIAGNOSTICS_CREATE_REPORT_COMPLETED";
+export const STATEMENT_DIAGNOSTICS_CREATE_REPORT_FAILED =
+  "ADMIN_UI/STATEMENT_DIAGNOSTICS_CREATE_REPORT_FAILED";
+export const OPEN_STATEMENT_DIAGNOSTICS_MODAL =
+  "ADMIN_UI/OPEN_STATEMENT_DIAGNOSTICS_MODAL";
+
+export type StatementDiagnosticsReportsResponse = cockroach.server.serverpb.StatementDiagnosticsReportsResponse;
+
+export type RefreshStatementsDiagnostics = Action<
+  typeof STATEMENTS_DIAGNOSTICS_REFRESH
+>;
+
+export type RequestStatementsDiagnostics = Action<
+  typeof STATEMENTS_DIAGNOSTICS_REQUEST
+>;
+
+export type RequestStatementsDiagnosticsFailed = ActionWithPayload<
+  typeof STATEMENTS_DIAGNOSTICS_FAILED,
+  Error
+>;
+
+export type RequestStatementsDiagnosticsReceived = ActionWithPayload<
+  typeof STATEMENTS_DIAGNOSTICS_RECEIVED,
+  StatementDiagnosticsReportsResponse
+>;
+
+export type StatementDiagnosticsInvalidated = Action<
+  typeof STATEMENTS_DIAGNOSTICS_INVALIDATED
+>;
+
+export type CreateStatementsDiagnosticsReport = ActionWithPayload<
+  typeof STATEMENT_DIAGNOSTICS_CREATE_REPORT,
+  string
+>;
+
+export type CreateStatementsDiagnosticsReportComplete = Action<
+  typeof STATEMENT_DIAGNOSTICS_CREATE_REPORT_COMPLETED
+>;
+
+export type CreateStatementsDiagnosticsReportFailed = Action<
+  typeof STATEMENT_DIAGNOSTICS_CREATE_REPORT_FAILED
+>;
+
+export type OpenStatementsDiagnosticsModal = ActionWithPayload<
+  typeof OPEN_STATEMENT_DIAGNOSTICS_MODAL,
+  string
+>;
+
+export type StatementsDiagnosticsActions =
+  | RefreshStatementsDiagnostics
+  | RequestStatementsDiagnostics
+  | RequestStatementsDiagnosticsFailed
+  | RequestStatementsDiagnosticsReceived
+  | StatementDiagnosticsInvalidated
+  | CreateStatementsDiagnosticsReport
+  | OpenStatementsDiagnosticsModal
+  | CreateStatementsDiagnosticsReportComplete
+  | CreateStatementsDiagnosticsReportFailed;
+
+export interface StatementDiagnosticsState {
+  data: StatementDiagnosticsReportsResponse;
+  lastError: Error;
+  valid: boolean;
+}
+
+const initialState: StatementDiagnosticsState = {
+  data: null,
+  valid: true,
+  lastError: null,
+};
+
+export const statementsDiagnosticsReducer: Reducer<
```

Comment:
I do have some concerns about using `redux-toolkit` but understand that it might reduce boilerplate and code repetition. The main concern is that reducer is supposed to be a simple `reducer function` + `switch ... case`, very simple building blocks. And `createReducer` just wraps it in its own API.
Also, I would say, that Typescript types/interfaces produce most of the boilerplate, and `redux-toolkit` won't fix it.
I think it's would be interesting to compare current implementation and the same definitions using `redux-toolkit`

Reply:
Sounds good, thx for the thoughts. Let's keep things as-is for now.